src/gui/uithememanager.cpp

```diff
@@ -58,6 +61,90 @@ namespace
 
         return {};
     }
+
+    QByteArray readFile(const QString &fileName)
+    {
+        QFile file {fileName};
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+        {
+            LogMsg(UIThemeManager::tr("UITheme - Failed to open \"%1\". Reason: %2")
+                    .arg(QFileInfo(fileName).fileName(), file.errorString())
+                   , Log::WARNING);
+            return {};
+        }
+
+        return file.readAll();
+    }
+
+    class QRCThemeSource final : public UIThemeSource
+    {
+    public:
+        QByteArray styleSheet() override
+        {
+            return readFile(m_qrcThemeDir + STYLESHEET_FILE_NAME);
+        }
+
+        QByteArray config() override
+        {
+            return readFile(m_qrcThemeDir + CONFIG_FILE_NAME);
+        }
+
+        QString iconPath(const QString &iconId) override
+        {
+            return findIcon(iconId, m_qrcIconsDir);
+        }
+
+    private:
+        const QString m_qrcThemeDir {":/uitheme/"};
```

Comment:
Are you going to use `STYLESHEET_RESOURCES_DIR`?

Reply:
they are logically different.

Reply:
It seems like it would be nice to have explanations of what each constant refers to.